mantle/platform/api/gcloud/image.go

```diff
@@ -154,3 +154,21 @@ func (a *API) DeleteImage(name string) (*Pending, error) {
 	opReq := a.compute.GlobalOperations.Get(a.options.Project, op.Name)
 	return a.NewPending(op.Name, opReq), nil
 }
+
+func (a *API) UpdateImage(spec *ImageSpec) (*Pending, error) {
+
+    // Only the following fields can be modified:
+    //      family, description, deprecation status
```

Comment:
Given this `Patch` API can modify deprecation status any reason to not switch `deprecate-image` to utilize it and drop the old way to consolidate on a single gcloud endpoint?

Reply:
I thought about it but leaned this way because with deprecate you can provide a deprecation state and a replacement image to point to. Adding both of those args here I thought might be a bit confusing. Since we already had `deprecate-image` I decided to just use that rather than adding it over here too. I'm kind of on the fence about leaving them split or combining them, so help me decide :)

Reply:
Hmm, the lack of ability to add a replacement image is a good point. If we're using that functionality and this doesn't have parity then I'm fine with leaving it on the current method.

Reply:
To be clear, I do think you can add a replacement image with `update-image` as well. I was mostly saying that adding the extra option may have been confusing to the user so since I already had `deprecate-image` I decided to just go with that.

If we were going to update `update-image` I think the new set of options would be:

- `ore gcloud update-image`
- `[--family=ImageFamily]`
- `[--description=ImageDescription]`
- `[--state=DeprecationState]`
- `[--replacement=DeprecationReplacement]`

So I'd add those other options and then just make sure the help text tried to explain they were for image deprecation purposes.

Reply:
I'm fine with the `update-image` ore command not providing a parameter for setting the deprecation state if you want to maintain that distance.

I'm speaking about this `UpdateImage` API function having support for it and switching the `deprecate-image` ore command to use this function instead of the current one.

Reply:
ok i'll look at this when I get back to the code tonight or tomorrow and try to figure out the best path forward. I'll either update the deprecate to call UpdateImage API or I'll just axe the deprecate-image file and just use update-image for everything.

Reply:
https://github.com/coreos/coreos-assembler/pull/1444#issuecomment-628921354